lib/smart_proxy_remote_execution_ssh/plugin.rb

```diff
@@ -20,9 +21,45 @@ class Plugin < Proxy::Plugin
                      # :mqtt_port               => nil,
                      :mode                    => :ssh
 
+    load_validators ssh_log_level: Proxy::RemoteExecution::Ssh::Validators::SshLogLevel,
+                    rex_ssh_mode: Proxy::RemoteExecution::Ssh::Validators::RexSshMode
+
+    load_programmable_settings do |settings|
+      if settings[:ssh_identity_key_file] && !settings[:ssh_identity_public_key_file]
+        settings[:ssh_identity_public_key_file] = "#{settings[:ssh_identity_key_file]}.pub"
+      end
+
+      if settings[:async_ssh]
+        Plugin.logger.warn('Option async_ssh is deprecated, use ssh-async mode instead.')
+
+        case setting_value
+        when :ssh
+          Plugin.logger.warn('Deprecated option async_ssh used together with ssh mode, switching mode to ssh-async.')
+          settings[:mode] = :'ssh-async'
+        when :'async-ssh'
+          # This is a noop
+        else
+          Plugin.logger.warn('Deprecated option async_ssh used together with incompatible mode, ignoring.')
+        end
+      end
+
+      settings[:mode] = settings[:mode].to_sym
+      settings[:ssh_log_level] = settings[:ssh_log_level].to_sym
+    end
+
+    validate_readable :ssh_identity_key_file, :ssh_identity_public_key_file
```

Comment:
Am I reading it right that this doesn't expand  the path?

Reply:
No, it doesn't. Perhaps it needs to be in `load_programmable_settings` instead of `lib/smart_proxy_remote_execution_ssh.rb`?